packages/flutter/lib/src/gestures/multidrag.dart

```diff
@@ -223,6 +223,7 @@ abstract class MultiDragGestureRecognizer extends GestureRecognizer {
     )
     super.kind,
     super.supportedDevices,
+    this.allowedMouseButtons,
```

Comment:
This seems to conflict with the `supportedDevices` parameter above it. What if it does not include `PointerDeviceKind.mouse`?

Reply:
That's a good question. Shall I make `supportedDevices == mouse && ...`?

Reply:
I am not sure. In general, having both parameters may be a bit unclear as far as the API surface goes.

What about other gestures? Or other places in the framework where we handle mouse buttons? It feels like we might need a more abstract design that can satisfy a consistent pattern across the framework.

Reply:
Yeah, I thought about this. The `buttons` API is also "very weird" because it uses bit-wise operations while everything else uses enum + set.

I just, well... don't know how to propose something better. Ideally it would be a sub-property from mouse, so you could have `mouse(clickType: primary | secondary | scroll | others)`, then unfold the mouse property when mouse is being used, but I don't think Dart supports anything like this (seems more like a sealed class thing?).

Reply:
You can right-click with `stylus`, and in some cases `touch` depending on iPadOS Info.plist. Probably the field name should be changed to `allowedButtons`. But on `Draggable` it could be kept as `allowedMouseButtons`, since in that context `allowedButtons` makes me think of a button widget, and since the parameter type is `int`, it is not really self-documenting.

Reply:
`typedef mouseButtons = int`. A possible way that could make allowedMouseButtons easier to understand.

Reply:
Making this specific to mice excludes the other valid inout devices listed here, like stylus. What about `inputDeviceModifiers` or something along those lines? Keeping it abstract will be inclusive of all input devices, and avoid confusion with the term `buttons`.

Reply:
Sure, we could have something like that. But `allowedMouseButtons` that's executed from behind the scenes is very "obscure"/low-level. It uses the bit-wise operations and so... But I'll try a better name, let's see.

Reply:
I changed from `allowedMouseButtons` to `allowedInputPointers`, but Flutter uses buttons in many places, like here:
<img width="573" alt="image" src="https://user-images.githubusercontent.com/351125/194918169-8f0eadb1-4137-429c-9084-ca2087f5de5f.png">